### PR TITLE
Show message in conflicts

### PR DESCRIPTION
**Pull request by jvcoutinho**
Solves #70.

Adds a parameter that allows the program to output messages in conflicts.

Below are listed the conflicts affected by this parameter and their outputs:

- When a new element A refers an edited one B: **"added A that refers to edited B"**
- Mutual renaming of method m(): **"mutual renaming of m() to different signatures"**
- Deletion of class A with a new instance of its renamed counterpart: **"possible deletion of class A"**
- Imports' conflict: **"type ambiguity error"** or **"possible behavioral error"**

Currently, textual merge's conflicts are not affected by this.
And there is still an issue of how the signature of methods are displayed.

### CODE REVIEW

File: src/main/java/br/ufpe/cin/app/JFSTMerge.java [hunk 1]
```diff
@@ -82,6 +82,9 @@
             converter = RenamingStrategyConverter.class)
 	public static RenamingStrategy renamingStrategy = RenamingStrategy.SAFE;
 
+	@Parameter(names = "-m", description = "Shows some details of merge conflicts on their occurrences.")
```
Comment (pauloborba) [inline src/main/java/br/ufpe/cin/app/JFSTMerge.java]: Shows extra messages detailing conflict causes

File: src/main/java/br/ufpe/cin/mergers/handlers/DeletionsHandler.java [hunk 2]
```diff
@@ -193,9 +193,9 @@ private static void generateNonTerminalConflict(FSTNode source,
 			FSTNonTerminal declarationInSource = correspondingInSource.getParent();
 			MergeConflict newConflict;
 			if(isLeftDeletion){
-				newConflict = new MergeConflict(null, baseNode, declarationInSource);
+				newConflict = new MergeConflict(null, baseNode, declarationInSource, "possible deletion of class " + baseNode.getName());
```
Comment (pauloborba) [inline src/main/java/br/ufpe/cin/mergers/handlers/DeletionsHandler.java]: rename

Comment (pauloborba) [reply src/main/java/br/ufpe/cin/mergers/handlers/DeletionsHandler.java]: and refer new classes

File: src/main/java/br/ufpe/cin/mergers/util/RenamingUtils.java [hunk 3]
```diff
@@ -110,7 +110,8 @@ public static boolean haveSameParent(FSTNode left, FSTNode right) {
     }
 
     public static void generateMutualRenamingConflict(MergeContext context, FSTNode leftNode, FSTNode baseNode, FSTNode rightNode, FSTNode mergeNode) {
-        MergeConflict conflict = new MergeConflict(leftNode, baseNode, rightNode);
+        String conflictMessage = new StringBuilder("mutual renaming of ").append(baseNode.getName()).append(" to different signatures").toString();
```
Comment (pauloborba) [inline src/main/java/br/ufpe/cin/mergers/util/RenamingUtils.java]: poderia ter sido deletado. verifica o doc com a discussao.

Comment (jvcoutinho) [reply src/main/java/br/ufpe/cin/mergers/util/RenamingUtils.java]: não, deleções mutuas não dão conflito

Comment (pauloborba) [reply src/main/java/br/ufpe/cin/mergers/util/RenamingUtils.java]: @jvcoutinho mas um nao poderia ter sido deletado e o outro renomeado?

**Comment by guilhermejccavalcanti** [comment]
a tabela de renaming diz que sempre que há renaming + deleção deve dar
conflito. A implementação tá assim, João?

Em seg, 30 de set de 2019 às 14:00, Paulo Borba <notifications@github.com>
escreveu:

**Comment by jvcoutinho** [comment]
Está sim. Foi erro meu.

**Comment by guilhermejccavalcanti** [comment]
@jvcoutinho  o que falta aqui?

File: src/main/java/br/ufpe/cin/mergers/util/RenamingUtils.java (hunk 3 above)
Comment (pauloborba) [reply src/main/java/br/ufpe/cin/mergers/util/RenamingUtils.java]: @jvcoutinho ?

Comment (jvcoutinho) [reply src/main/java/br/ufpe/cin/mergers/util/RenamingUtils.java]: Sim, havia respondido embaixo. Eu mudei as mensagens.

**Pull request merged**